walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,170 @@
+
+    // Some type of collection to store all of the (unsorted) unavailable meeting times. 
+    Collection<TimeRange> unavailableTimesSet = new HashSet<>();
+
+    // Check to see if the duration is longer than a day.
+    if (request.getDuration() == TimeRange.WHOLE_DAY.duration() + 1) {
+      return availableTimes;
+    }
+
+    /* For each of the requested attendees, loop through all of the events. For each event,
+     * If that particular required attendee is attending it, add that meeting time slot to 
+     * The collection of unavailable meeting times. */
+
+    // Iterate over the HashSet of attendees in request.attendees. 
+    for (String requiredAttendee : request.getAttendees()) {
+      // Loop through all of the events that are going on that day.
+      for (Event currEvent : events) {
+        // Check to see if requiredAttendee is among the attendees at this particular event.
+        if (currEvent.getAttendees().contains(requiredAttendee)) {
+          unavailableTimesSet.add(currEvent.getWhen());
+        }
+      }
+    } 
+
+    /* Sort all the unavailabe meeting times (i.e TimeRanges) in order of start time
+     * (earliest start time first). */
+    List<TimeRange> unfixedSortedUnavailableTimes = new ArrayList<>();
+    unfixedSortedUnavailableTimes.addAll(unavailableTimesSet);
+    Collections.sort(unfixedSortedUnavailableTimes, TimeRange.ORDER_BY_START);
+
+    /* See if there's overlap in the unavailable times and fix them in new array. */
+    List<TimeRange> sortedUnavailableTimes = new ArrayList<>();
+
+    // if 0, last unavailable time in unfixedSortedUnavailableTimes didn't overlap with any before it
+    int statusFlag = 0; 
+    int curr = 0;
+    int next = curr + 1;
+    TimeRange currRange;
+    TimeRange nextRange;
+    if (unfixedSortedUnavailableTimes.size() > 0) {
+      while (next != unfixedSortedUnavailableTimes.size()) {
+        currRange = unfixedSortedUnavailableTimes.get(curr);
+        nextRange = unfixedSortedUnavailableTimes.get(next);
+    
+        if (!currRange.overlaps(nextRange)) {
+          sortedUnavailableTimes.add(currRange);
+          curr++;
+          next++;
+          continue;
+        }
+
+        while (currRange.overlaps(nextRange)) {
+          next++;
+          if (next == unfixedSortedUnavailableTimes.size()) {
+            break;
+          } else {
+            nextRange = unfixedSortedUnavailableTimes.get(next);
+          }
+        }
+
+        // If next is now the size of unfixedSortedUnavailableTimes, currRange overlaps 
+        // With all of the following events.
+        if (next == unfixedSortedUnavailableTimes.size()) {
+          statusFlag = 1;
+        
+          next--;
+          // reset back to the most recent range that overlap.
+          nextRange = unfixedSortedUnavailableTimes.get(next); 
+        
+          // The current range fully contains the last thing it overlaps with.
+          if (currRange.contains(nextRange)) {
```

Comment:
Maybe you can just use max(currRange.end(), nextRange.end())? Or create your own function to combine the two range. For example:
sortedUnavailableTimes.add(currRange + nextRange);

Reply:
I'm confused by what you're asking. To use max(currRang.end(), nextRange.end()) for what?